src/connector/src/sink/elasticsearch_opensearch/elasticsearch_opensearch_common.rs

```diff
@@ -0,0 +1,594 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use futures::{FutureExt, TryFuture};
+use itertools::Itertools;
+use risingwave_common::array::{Op, StreamChunk};
+use risingwave_common::catalog::Schema;
+use risingwave_common::row::Row;
+use risingwave_common::types::DataType;
+use serde::Deserialize;
+use serde_json::{json, Map, Value};
+use serde_with::{serde_as, DisplayFromStr};
+use url::Url;
+use with_options::WithOptions;
+
+use super::super::encoder::template::TemplateEncoder;
+use super::super::encoder::{JsonEncoder, RowEncoder};
+use super::super::SinkError;
+use super::elasticsearch::ES_SINK;
+use super::opensearch::OPENSEARCH_SINK;
+use crate::sink::log_store::DeliveryFutureManagerAddFuture;
+use crate::sink::writer::AsyncTruncateSinkWriter;
+use crate::sink::Result;
+
+pub const ES_OPTION_DELIMITER: &str = "delimiter";
+pub const ES_OPTION_INDEX_COLUMN: &str = "index_column";
+pub const ES_OPTION_INDEX: &str = "index";
+pub const ES_OPTION_ROUTING_COLUMN: &str = "routing_column";
+
+#[serde_as]
+#[derive(Deserialize, Debug, Clone, WithOptions)]
+pub struct ElasticSearchOpenSearchConfig {
```

Comment:
This file is too large. Can we split it into smaller files? We can split it into several files for config, formatter, and client respectively.